src/main/scala/castalia/StubService.scala

```diff
@@ -1,43 +1,64 @@
 package castalia
 
+import akka.actor.ActorSystem
 import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
 
-/**
-  * Created by Jens Kat on 27-11-2015.
-  */
-trait StubService extends BaseService {
+class StubService(theStubsByEndpoints: StubConfigsByEndpoint)(implicit val system: ActorSystem) extends Routes {
+  protected val serviceName = "StubRoutes"
 
-  protected val serviceName = "StubService"
+  protected def stubsByEndpoints: StubConfigsByEndpoint = theStubsByEndpoints
 
-  val stubRoutes = pathPrefix("stubs") {
-    handleRejections(totallyMissingHandler) {
-      path("hardcodeddummystub") {
-        get {
-          complete("Yay!")
-        }
-      } ~
-      path("anotherstub") {
-        get {
-          complete("Different response")
-        }
-      } ~
-      pathPrefix("dynamicdummystub"){
-        path("default") {
-          parameter("response") { anyString =>
-              get {
-                complete(Response(None, anyString))
-              }
-          }
-        } ~
-        path(IntNumber){ id =>
-          parameter("response") { anyString =>
-            get {
-              complete(Response(Some(id), anyString))
+  protected lazy val dynamicStubRoutes = {
+    def createRoute(endpoint: String, responses: ResponsesByRequest): Route = path(endpoint / Segment) { id =>
+      get {
+        val response: Option[StubResponse] = responses.get(id)
+
+        response match {
+          case Some((statusCode: StatusCode, optResponse: AnyJsonObject)) =>
+            optResponse match {
```

Comment:
`optResponse.orElse("") map { complete(statusCode, ) }` (not checked, but something like this would do the same as the pattern match